bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcarePetHandler.java

```diff
@@ -0,0 +1,251 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.surepetcare.internal.handler;
+
+import static org.openhab.binding.surepetcare.internal.SurePetcareConstants.*;
+
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.quantity.Mass;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.surepetcare.internal.SurePetcareAPIHelper;
+import org.openhab.binding.surepetcare.internal.SurePetcareApiException;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcareDevice;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcareHousehold;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcarePet;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcarePetActivity;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcarePetFeeding;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcareTag;
+import org.openhab.binding.surepetcare.internal.utils.ByteArrayFileCache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SurePetcarePetHandler} is responsible for handling the things created to represent Sure Petcare pets.
+ *
+ * @author Rene Scherer - Initial Contribution
+ * @author Holger Eisold - Added pet feeder status, location time offset
+ */
+@NonNullByDefault
+public class SurePetcarePetHandler extends SurePetcareBaseObjectHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SurePetcarePetHandler.class);
+
+    private static final String JPEG_CONTENT_TYPE = "image/jpeg";
+
+    private static final ByteArrayFileCache IMAGE_CACHE = new ByteArrayFileCache("org.openhab.binding.surepetcare");
```

Comment:
Would it maybe make more sense to just store the actual images in the folder, that seems more optimal to me?

Reply:
I've used this way because we dont have to take care of that specific folder... and if a user deletes the cache... all "old images" are gone... I'd like to keep this... (update to newer version as mentioned below)

If you want me to change this... do you have a binding in mind where a can take a look how to implement this?